crates/solidity/inputs/language/src/dsl.rs

```diff
@@ -0,0 +1,915 @@
+use codegen_grammar::*;
+
+slang_grammar! {
+
+    config = {
+
+        name = "Solidity" ;
+
+        versions = [
+            "0.4.11" , "0.4.12" , "0.4.13" , "0.4.14" , "0.4.15" , "0.4.16" , "0.4.17" , "0.4.18" , "0.4.19" , "0.4.20" , "0.4.21" , "0.4.22" ,
+            "0.4.23" , "0.4.24" , "0.4.25" , "0.4.26" ,
+
+            "0.5.0" , "0.5.1" , "0.5.2" , "0.5.3" , "0.5.4" , "0.5.5" , "0.5.6" , "0.5.7" , "0.5.8" , "0.5.9" , "0.5.10" , "0.5.11" , "0.5.12" ,
+            "0.5.13" , "0.5.14" , "0.5.15" , "0.5.16" , "0.5.17" ,
+
+            "0.6.0" , "0.6.1" , "0.6.2" , "0.6.3" , "0.6.4" , "0.6.5" , "0.6.6" , "0.6.7" , "0.6.8" , "0.6.9" , "0.6.10" , "0.6.11" , "0.6.12" ,
```

Comment:
If we are in `lexical context Default`, and the parser encounters a production/parser from a different context, for example `lexical context VersionPragma`, my understanding is that it should treat it as a sub-language, is that correct? That means that while parsing `VersionPragma` (and its children) it should only try to scan tokens defined in that context. But I see it referencing scanners that are outside of this context (for example `VersionPragmaValue`).

How does that work? Not sure I understand how the boundary between these contexts is defined.

Reply:
It could be used to group scanners, but the issue is that scanners are shared between lexical contexts. A lexical context is just a set of scanners that are referenced by one or more parsers, which is why it is a grouping of parsers. The system can switch between lexical contexts on a per-parser basis i.e. the context is not an inherited/transitive property.

This avoids having to duplicate scanners for each sub-language.